cloud-consumption/src/test/java/com/sequenceiq/consumption/job/storage/StorageConsumptionJobServiceTest.java

```diff
@@ -0,0 +1,201 @@
+package com.sequenceiq.consumption.job.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.springframework.context.ApplicationContext;
+
+import com.sequenceiq.cloudbreak.quartz.model.JobResource;
+import com.sequenceiq.consumption.configuration.repository.ConsumptionRepository;
+import com.sequenceiq.consumption.domain.Consumption;
+import com.sequenceiq.consumption.service.ConsumptionService;
+
+@ExtendWith(MockitoExtension.class)
+public class StorageConsumptionJobServiceTest {
+
+    @Mock
+    private ApplicationContext applicationContext;
+
+    @Mock
+    private Scheduler scheduler;
+
+    @Mock
+    private StorageConsumptionConfig storageConsumptionConfig;
+
+    @Mock
+    private ConsumptionService consumptionService;
+
+    @Mock
+    private ConsumptionRepository consumptionRepository;
+
+    @InjectMocks
+    private StorageConsumptionJobService underTest;
+
+    @Captor
+    private ArgumentCaptor<JobDetail> jobDetailCaptor;
+
+    @Captor
+    private ArgumentCaptor<JobKey> jobKeyCaptor;
+
+    @BeforeEach
+    public void setUp() {
+        lenient().when(applicationContext.getBean(ConsumptionRepository.class)).thenAnswer(interaction -> consumptionRepository);
+        lenient().when(storageConsumptionConfig.getIntervalInMinutes()).thenAnswer(invocation -> 30);
```

Comment:
Why not `thenReturn()`? `thenAnswer()` is typically employed when the answer depends on the actual arg values of the called method, or when the answer is constructed dynamically.